### ISSUE TITLE
Please remove pdfminer2 dependency

### ISSUE THREAD

**Issue by kpine** [beancount/beancount#788]
I have been using custom PDF importers in Beancount with [pdfminer-six](https://github.com/pdfminer/pdfminer.six). After upgrading to Beancount 2.3.6 my importers started failing. This is because Beancount 2.3.6 has [added](https://github.com/beancount/beancount/commit/04dab637af61de51614d9577e569869a497f733c) pdfminer2 as a dependency. As the APIs are not compatible, the python imports fail.

I can workaround this by uninstalling pdfminer2 and installing pdfminer-six afterwards, but since there doesn't actually seem to be any real dependency on pdfminer2, is it necessary to include it?

**Comment by dnicolodi** [comment]
pdfminer2 is used in some tests. I agree that it would be better to do not install it if not needed. But I don't understand the relation between beancount depending on pdfminer2 and your importers depending on pdfminer-six, are these two packages using the same Python namespace?

**Comment by kpine** [comment]
> But I don't understand the relation between beancount depending on pdfminer2 and your importers depending on pdfminer-six, are these two packages using the same Python namespace?

Yes, they both are used via `import pdfminer`.

It seems [pdfminer](https://github.com/euske/pdfminer) is the original project. That project has been unmaintained since 2020. They recommend using [pdfminer-six](https://github.com/pdfminer/pdfminer.six) as a replacement fork, which is probably how I ended up there.

I'm not even sure what [pdfminer2](https://github.com/metachris/pdfminer) is supposed to be? According to github, pdfminer2 is a fork of pdfminer-six from 2015, with [no functional changes](https://github.com/pdfminer/pdfminer.six/compare/master...metachris:pdfminer:master), only a project name change. So pdfminer2's last updates are older than both the original (pdfminer), and its fork (pdfminer-six). It really just looks like the start of a planned fork that was abandoned.

> pdfminer2 is used in some tests.

Maybe you can clarify that. Is anyone/anything running these tests? I see `pdf2txt.py` is called from the example tests. However, the [CI job](https://github.com/beancount/beancount/blob/v2/.github/workflows/linux-ci.yaml) only runs `make test` which does not include the example tests. To run the example tests would require one of the variants of `make vtest`, which I don't see running? Another thing is that the CI job does also not install the beancount module, it installs deps via `requirements_dev.txt`, which is missing pdfminer2. Am I missing where these are run?

If you install pdfminer2 (or beancount) and actually try to run the tests, you'll see they are skipped. Apparently pdfminer2 ships a broken `pdf2txt.py` script:

```console
$ pdf2txt.py --help
bash: /tmp/beancount/venv/bin/pdf2txt.py: cannot execute: required file not found
$ head -n1 $(which pdf2txt.py)
#!/Users/chris/Projects/chris/various/pdfminer/venv/bin/python
```

If you install pdfminer-six instead, the tests will not be skipped and will pass.

Last thing, I went to look at what the v3 branch is doing. Coincidentally, it appears you actually [replaced](https://github.com/beancount/beangulp/commit/c32d0fe81aa19a71a0062eb6ed147dd462e27923) pdfminer2 with pdftotext in the beangulp repo. :laughing: Maybe I should do the same as pdfminer-six is having its own maintenance problems.

**Comment by dnicolodi** [comment]
Thanks for looking into this @kpine What you report sounds vaguely familiar. I think I discovered most of this at the time I switched beangulp to use `pdftotext` but then forgot about it. @blais actually added the dependency to beancount. I trusted him on the choice. Anyhow, I've opened a PR that moves pdfminer to be an optional dependency, switched to pdfminer-six, and actually runs the example tests in CI.
